projects/storefrontapp-e2e-cypress/cypress/integration/b2b/regression/unit-level-order/b2b-unit-level-order-history.e2e-spec.ts

```diff
@@ -0,0 +1,203 @@
+      cy.get('[class=ng-option-label]').contains('Unit (Descending)').click();
+      cy.wait(ordersToReload);
+      checkThatOrdersAreSorted(fieldUnit, false);
+    });
+  });
+
+  describe('Check presence and  order of sorting option', () => {
+    it('should display unit level orders page and drop down menu with sorting options', () => {
+      cy.get('.ng-input').first().click();
+      cy.get('.ng-option')
+        .then(($items) => {
+          return $items.map((_index, html) => Cypress.$(html).text()).get();
+        })
+        .should('deep.eq', [
+          'Date',
+          'Order Number',
+          'Buyer (Ascending)',
+          'Buyer (Descending)',
+          'Unit (Ascending)',
+          'Unit (Descending)',
+        ]);
+    });
+  });
+
+  describe('Check unit-level orders filtering', () => {
+    it('should filter order history by buyer name', () => {
+      const ordersToReload = registerReloadOrdersAlias(orderHistoryPageUrl);
+      cleanFilter(filterUnit).wait(ordersToReload);
+      setFilter(filterBuyer, buyerGiSun).wait(ordersToReload);
+      checkThatOrdersAreFilteredBy(fieldBuyer, buyerGiSun);
+      setFilter(filterBuyer, buyerMarkRivers).wait(ordersToReload);
+      checkThatOrdersAreFilteredBy(fieldBuyer, buyerMarkRivers);
+      setFilter(filterBuyer, buyerWilliamHunter).wait(ordersToReload);
+      checkThatOrdersAreFilteredBy(fieldBuyer, buyerWilliamHunter);
+    });
+
+    it('should filter order history by unit name', () => {
+      const ordersToReload = registerReloadOrdersAlias(orderHistoryPageUrl);
+      cleanFilter(filterBuyer).wait(ordersToReload);
+      setFilter(filterUnit, unitCustomRetail).wait(ordersToReload);
+      checkThatOrdersAreFilteredBy(fieldUnit, unitCustomRetail);
+      setFilter(filterUnit, unitServicesWest).wait(ordersToReload);
+      checkThatOrdersAreFilteredBy(fieldUnit, unitServicesWest);
+    });
+
+    it('should not display orders history when mutually exclusive filers values are set', () => {
+      setMutuallyExclusiveFilterValues();
+      checkThatOrderListIsEmpty();
+    });
+
+    it('should clear the filter after the reset icon is clicked', () => {
+      resetFilter(filterBuyer);
+      resetFilter(filterUnit);
+      checkThatFilterIsCleaned(filterBuyer);
+      checkThatFilterIsCleaned(filterUnit);
+      checkThatOrdersListIsNotEmpty();
+    });
+  });
+
+  function registerReloadOrdersAlias(urlToWaitFor: string) {
+    return interceptGet('waitForOrderHistoryToReload', urlToWaitFor);
+  }
+
+  function checkThatOrderListIsEmpty() {
+    cy.get('#order-history-table').should('not.exist');
+  }
+
+  function checkThatOrdersListIsNotEmpty() {
+    cy.get('#order-history-table').should((tableData) => {
+      expect(tableData).to.not.be.empty;
+    });
+  }
+
+  function checkThatOrdersAreSorted(
+    dataCellSelector: string,
+    ascending: boolean = true
+  ) {
+    cy.get(dataCellSelector, { timeout: 500 }).then((items) => {
```

Comment:
Why does the timeout have to be reduced?

Reply:
ups, it is my oversight, I have tested this options and left it unintentionally
updated